src/App.tsx

```diff
@@ -1,14 +1,53 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { Todo } from './types/Todo';
+import './App.css';
+import { getTodos } from './api';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { FilterType } from './types/FilterType';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completeTodoFilter, setCompleteTodoFilter] = useState('all');
+  const [query, setQuery] = useState('');
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+
+  const prettyQuery = query.toLowerCase().trim();
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  let visibleTodos = [...todos];
+
+  switch (completeTodoFilter) {
+    case (FilterType.active):
+      visibleTodos = [...todos].filter(el => el.completed === false);
+      break;
```

Comment:
It's better to use [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) to save visibleTodos